### PR TITLE
Added precise types for ast Constants

### LINKED ISSUE #4005: mypy can't decide if ast.Str.s is a string or not, depending on Python 3.7 vs 3.8
* This is a bug (I think)

```python
import ast
from typing import Optional


def eval_join(node: ast.AST) -> Optional[str]:
    if (isinstance(node, ast.Call) and isinstance(node.func, ast.Attribute)
            and isinstance(node.func.value, ast.Str)
            and node.func.attr == 'join'):
        return node.func.value.s    # <-- mypy --strict complains here
    return None
```

When I run mypy --strict on this file on Python 3.8, I get

```
ast_is_a_string.py:9: error: Returning Any from function declared to return "Optional[str]"
Found 1 error in 1 file (checked 1 source file)
```

When I run mypy --strict on this file on Python 3.6 or 3.7, I get no errors.

If I wrap the return value in a `typing.cast(typing.Str, node.func.value.s)`, mypy on 3.8 becomes happy, but mypy on 3.6 or 3.7 complains about redundant cast to "str".

* What is the behavior/output you expect?

I expect mypy to behave the same on Python 3.6 through 3.8.

* What are the versions of mypy and Python you are using?

mypy 0.770, Python 3.6 through 3.8.

* Do you see the same issue after installing mypy from Git master?

Yes

* What are the mypy flags you are using? (For example --strict-optional)

Just --strict.

### PR DESCRIPTION
**Pull request by rohitkg98**
Fixes #4005

### CODE REVIEW

File: stdlib/3/ast.pyi [hunk 1]
```diff
@@ -14,9 +14,21 @@ from typing_extensions import Literal
 from _ast import *  # type: ignore
 
 if sys.version_info >= (3, 8):
-    class Num(Constant): ...
-    class Str(Constant): ...
-    class Bytes(Constant): ...
+    class Num(Constant):
+        value: complex
```
Comment (JelleZijlstra) [inline stdlib/3/ast.pyi:18]: Float isn't a subtype of complex, right? In practice it's much more commonly a float or int.

File: stdlib/3/ast.pyi [hunk 2]
```diff
@@ -14,9 +14,21 @@ from typing_extensions import Literal
...
+        kind: Optional[str]
+        # Aliases for value, for backwards compatibility
+        s: Any
```
Comment (JelleZijlstra) [inline stdlib/3/ast.pyi:21]: This is `n`, not `s`

File: stdlib/3/ast.pyi (hunk 1 above)
Comment (rohitkg98) [reply stdlib/3/ast.pyi:18]: I was confused about that too, then I found this on mypy docs: https://mypy.readthedocs.io/en/stable/duck_type_compatibility.html#duck-type-compatibility

it says both `int` and `float` are duck types compatible with `complex`

File: stdlib/3/ast.pyi (hunk 2 above)
Comment (rohitkg98) [reply stdlib/3/ast.pyi:21]: `n` is inherited from `Constant` as a `complex` type already, so is `s` as `Any` type, I will remove the declaration

**Pull request merged**